### PR TITLE
distrobox: 1.6.0.1 -> 1.7.1

### LINKED ISSUE #296592: Update request: distrobox 1.6.0.1 → 1.7.0.1

### PR DESCRIPTION
**Pull request by pongo1231**
## Description of changes

https://github.com/89luca89/distrobox/releases/tag/1.7.1

closes #296592

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jmarmstrong1207** [comment]
@AtilaSaraiva

**Review by nadir-ishiguro — APPROVED** [review]
Works great for me.

File: pkgs/applications/virtualization/distrobox/default.nix [hunk 1]
```diff
@@ -2,23 +2,20 @@
...
+    hash = "sha256-g936ofLlYZTgzD7lKpjKT6ct6mAv7IJsgkaAUXXAJLw=";
   };
 
   dontConfigure = true;
   dontBuild = true;
 
   nativeBuildInputs = [ makeWrapper ];
 
-  # https://github.com/89luca89/distrobox/pull/1080
-  patches = [ ./always-mount-nix.patch ];
```
Comment (bubuntux) [inline pkgs/applications/virtualization/distrobox/default.nix:20]: we should probably delete this patch file from the repo since is no longer need it?

**Comment by bubuntux** [comment]
a new minor version just dropped https://github.com/89luca89/distrobox/releases/tag/1.7.0.1

**Comment by chewblacka** [comment]
This update doesn't seem to work for me. When running e.g.
```
distrobox create -i ubuntu:latest --name ubuntu
distrobox enter ubuntu
```
After container setup I get the error:
```
Error: can only create exec sessions on running containers: container state improper
```

**Comment by pongo1231** [comment]
Can replicate, commit [609c05](https://github.com/89luca89/distrobox/commit/609c053c7a321827f412aaf9e251c882389a9268) being the culprit after 1.7.0.

**Comment by pongo1231** [comment]
`findmnt -no SOURCE "/etc/localtime"` fails as /etc/localtime does not exist on NixOS by default. #1296 fixes this.

**Comment by chewblacka** [comment]
> `findmnt -no SOURCE "/etc/localtime"` fails as /etc/localtime does not exist on NixOS by default. [89luca89/distrobox#1296](https://github.com/89luca89/distrobox/pull/1296) fixes this.

Can confirm the patch fixes the bug and everything works as expected now.

**Comment by chewblacka** [comment]
1.7.1. is out now:
https://github.com/89luca89/distrobox/releases/tag/1.7.1

**Comment by pongo1231** [comment]
Updated.

**Review by chewblacka — APPROVED** [review]
LGTM

**Comment by NateWright** [comment]
Does anyone have this updated version working with podman? I am having this issue https://github.com/89luca89/distrobox/issues/1336#issue-2235655264

**Comment by chewblacka** [comment]
> Does anyone have this updated version working with podman? I am having this issue [89luca89/distrobox#1336 (comment)](https://github.com/89luca89/distrobox/issues/1336#issue-2235655264)

Strange. I followed your example and it seems to work fine for me.

**Comment by NateWright** [comment]
> > Does anyone have this updated version working with podman? I am having this issue [89luca89/distrobox#1336 (comment)](https://github.com/89luca89/distrobox/issues/1336#issue-2235655264)
> 
> Strange. I followed your example and it seems to work fine for me.
What version of podman are you using?

**Comment by NateWright** [comment]
> > What version of podman are you using?
> 
> podman version 4.9.3
> 
> Is there anything in directory `~/distrobox/home/u22` which might be interfering? I.e. does the same error occur in a new / clean directory?

No there shouldn't be. I tried the command you had above `distrobox create -i ubuntu:latest --name ubuntu` and `distrobox enter ubuntu` and it has the same issue.

**Pull request merged**